content/spin/kubernetes.md

```diff
@@ -0,0 +1,382 @@
+
+### Known Limitations
+
+- Each Pod will be constantly running it’s own HTTP listener which adds overhead vs Fermyon Cloud.
+- You can run containers and wasm modules on the same node, but you can't run containers and wasm modules on the same pod.
+
+### Setup
+
+Clone containerd shim repository. Cd into the directory and run make
+
+```bash
+git clone [https://github.com/deislabs/containerd-wasm-shims](https://github.com/deislabs/containerd-wasm-shims)
+cd containerd-wasm-shims
+make
+```
+
+Copy the `containerd-shim-spin-v1` to the `/bin` directory of kubernetes node image. Then add the following lines to the config.toml for containerd.
+
+```
+  [plugins.cri.containerd.runtimes.spin]
+    runtime_type = "io.containerd.spin.v1"
+```
+
+Create a file wasm-runtimeclass.yml and populate with the following information
+
+```yaml
+apiVersion: node.k8s.io/v1
+kind: RuntimeClass
+metadata:
+  name: "wasmtime-spin"
+handler: "spin"
+scheduling:
+  nodeSelector:
+    "kubernetes.azure.com/wasmtime-spin": "true"
+```
+
+Then register the runtime class with the cluster
+```yaml
+kubectl apply -f wasm-runtimeclass.yaml
+```
+
+{{ blockEnd }}
+
+## Run a Spin workload on Kubernetes
+### Introduction
+
+This guide demonstrates the commands to run a Spin workload in Kubernetes. It should apply to all Kubernetes varients which have been properly configured using one of the Setup guides.
+
+### Concepts
+
+Spin apps are bundled using either Bindle or OCI (as of Spin 0.8). These packages include the spin.toml file, and the wasm and static files which it references. While Kubernetes also uses OCI repositories, it expects the package to be in a container format.
+
+The containerd shim for Spin allows Kubernetes to appropriately schedule Spin applications as long as they have been wrapped in a lightweight “scratch” container.
+
+Once the Spin App Container has been created, it can be pushed to an OCI repository, and then deployed to an appropriately configured kubernetes cluster.
+
+### Requirements
+
+The current Spin k8s plugin relies on Docker and Kubectl under the hood. It’s important that both of these tools be installed, the Docker service be running, and KUBECONFIG environment variable be configured to point to a kubeconfig file for the desired Kubernetes cluster.
+
+### Install Plugin
+
+To install this plugin, simply run:
+
+```yaml
+spin plugin install -u https://raw.githubusercontent.com/chrismatteson/spin-plugin-k8s/main/k8s.json
+```
+
+### Workflow
+
+The workflow is very similar to the workflow for Fermyon Cloud. 
+
+The k8s plugin handles all of the tasks necessary to build the docker container, push it to a repository and deploy it into production.
+
+Just like with Fermyon Cloud, when something changes with the application, the workflow is to iterate the version in the spin.toml file, and restart the sequence from spin build.
+
+### Detailed Explanation of Steps
+
+#### spin new
```

Comment:
With all of these following commands, I would suggest adding each command into a code block (starting with a `$`) and also using the `<!-- @selectiveCpy -->` so that the user can click/copy and paste the commands (also might look neat using code blocks). For example, capitalise the heading, create a sentence that ends in a `:` before the code block begins and then utilise the code block.

![Screenshot 2023-04-14 at 12 35 48 pm](https://user-images.githubusercontent.com/9831342/231927343-95f7e158-7c85-48e2-9e34-c8bb443865f7.png)

The output will look something like this.

![Screenshot 2023-04-14 at 12 37 36 pm](https://user-images.githubusercontent.com/9831342/231927615-137963d0-6407-49d7-974a-1d8e625e7ea8.png)